### PR TITLE
fix(@schematics/angular): update or add module and target to non extend…

### LINKED ISSUE #14436: Older CLI projects don't match the tsconfig layout of newer projects
# 🐞 Bug report

### Command (mark with an `x`)

```
- [x] update
```

### Is this a regression?
No

### Description
Older projects (such as https://github.com/GDGTC/devfest/commits/master/src/tsconfig.json) have a single tsconfig file. This was never migrated by our tooling to `tsconfig.app.json` which means that some of our assumptions about these files are incorrect. This breaks things such as the schematics in 8 to take advantage of differential loading.

We should consider detecting these configs and migrating them.

### PR DESCRIPTION
**Pull request by alan-agius4**
…ed tsconfig

Fixes #14436

### CODE REVIEW

File: packages/schematics/angular/migrations/update-8/differential-loading.ts [hunk 1]
```diff
@@ -171,10 +128,37 @@ function getCompilerOptionsAstObject(host: Tree, tsConfigPath: string): JsonAstO
     return;
   }
 
+  const configExtends = findPropertyInAstObject(tsCfgAst, 'extends');
+  const isExtendedConfig = configExtends && configExtends.kind === 'string';
```
Comment (filipesilva) [inline packages/schematics/angular/migrations/update-8/differential-loading.ts]: @clydin asked this check to be stricter:

> Can we add some safety checks to the extends checks for the tsconfig?  If it extends from more than one or if the depth of extension is more than one, it should probably bail.
> a config that’s heavily customized will probably break if the migration starts modifying options

Comment (alan-agius4) [reply packages/schematics/angular/migrations/update-8/differential-loading.ts]: Let’s take this on slack, bailing out increases the changes of leaving broken projects as this migration is mandatory for other migrations such as for the lazy loading import syntax change.

**Comment by alexeagle** [comment]
@alan-agius4 this is too risky for 8 but @StephenFluin asks, could we find the tsconfig in the old location and update the module/target setting there?

**Comment by alan-agius4** [comment]
@alexeagle thats exactly what this PR does.

File: packages/schematics/angular/migrations/update-8/differential-loading.ts (hunk 1 above)
Comment (alexeagle) [reply packages/schematics/angular/migrations/update-8/differential-loading.ts]: sounds right to me - users can always look at the diff to understand what we did

**Pull request merged**